rust/src/lib.rs

```diff
@@ -1,21 +1,73 @@
 // SPDX-FileCopyrightText: © 2022 Svix Authors
 // SPDX-License-Identifier: MIT
 
+//! Rust client library for Svix.
+//!
+//! The main entry points of this library are the API client [`api::Svix`], and
+//! [`webhooks::Webhook`].
+
 #![warn(clippy::all)]
 #![forbid(unsafe_code)]
 
+use hyper::body::Bytes;
+use hyper_util::client::legacy::{connect::HttpConnector, Client as HyperClient};
+
 #[macro_use]
 extern crate serde_derive;
 #[macro_use]
 extern crate serde_repr;
 
 pub mod api;
 pub mod error;
+mod request;
 pub mod webhooks;
 
 #[rustfmt::skip]
-#[allow(dead_code)]
+#[allow(dead_code, clippy::all)]
 mod apis;
 #[rustfmt::skip]
-#[allow(dead_code)]
+#[allow(dead_code, clippy::all)]
 mod models;
+
+struct Configuration {
+    pub base_path: String,
+    pub user_agent: Option<String>,
+    pub client: HyperClient<Connector, http_body_util::Full<Bytes>>,
+    pub bearer_access_token: Option<String>,
+}
+
+// If no TLS backend is enabled, use plain http connector.
+#[cfg(not(any(feature = "native-tls", feature = "rustls-tls")))]
+type Connector = HttpConnector;
+
+// If only native TLS is enabled, use that.
+#[cfg(all(feature = "native-tls", not(feature = "rustls-tls")))]
+type Connector = hyper_tls::HttpsConnector<HttpConnector>;
+
+// If rustls is enabled, use that.
+#[cfg(feature = "rustls-tls")]
+type Connector = hyper_rustls::HttpsConnector<HttpConnector>;
```

Comment:
I wonder

- Should we support setups where TLS is completely disabled (at compile time)?
- If we were to not support the compile-time configuration, should we also enforce https at runtime (e.g. using `.https_only()` on the rustls connector builder)?
- If for some reason both native-tls and rustls are enabled at compile time, which one should we use? Currently this is rustls which is probably more secure and potentially avoids some compilation issues in particular around cross-compilation. It's less compatible with old TLS server implementations, but shouldn't matter unless somebody runs the API server behind an old reverse proxy.

Reply:
* I don't see any reason to support compile-time disabling of TLS -- that's not currently something we support, right?
* I don't think `https_only` is necessary -- it sounds prohibitive to anyone using this for local development. If we think it's an important option, that could potentially be a feature as well?
* I would default to rustls

Reply:
Yeah, compile-time disabling of TLS is definitely not a high-profile feature. It almost comes about naturally by both TLS backends being optional. If we wanted to have a proper error message for if both are disabled, that would take an approximate 5 lines of code and a small compile time hit from having a `build.rs` file, whereas the current solution of supporting it is 4 lines of code :smile:

Reply:
Ah, ok, I guess that makes sense. It's probably not a terrible idea to allow it then. But if we support one or the other by default, a user would have to go out of their way to do so though, right, i.e., by disabling default-features?

Reply:
Yeah, exactly. I've set `native-tls` as the default now, same as reqwest does but we could switch the default to rustls too.